lib/account.rb

```diff
@@ -0,0 +1,47 @@
+require 'csv'
+require 'time'
+
+module Bank
+  class Account
+    attr_reader :id, :balance, :date
+    def initialize(id, balance, date="nil")
+      @id = id
+      @balance = balance
+      @date = date
+      raise ArgumentError.new("balance must be >= 0") if balance < 0
+    end
+
+    def self.find(id1)
+      @accounts_array.each do |line|
```

Comment:
I like the idea of using the accounts loaded in `Account.all` here in `find`. But there's a problem - if `find` is run before `all`, you get an error! How could you guarantee that `all` has been run by the time you get here?